src/objective-c/GRPCClient/private/GRPCHost.m

```diff
@@ -98,6 +100,16 @@ - (nullable instancetype)initWithAddress:(NSString *)address {
       _address = address;
       _secure = YES;
       kHostCache[address] = self;
+
+      if (!connectivityMonitor) {
+        connectivityMonitor =
+            [GRPCConnectivityMonitor monitorWithHost:hostURL.host];
```

Comment:
Would that @synchronized(kHostCache) be good enough for concurrency? in both initWithAddress and flushChannelCache